stdlib/public/core/OutputStream.swift

```diff
@@ -330,9 +330,19 @@ internal struct _Stdout : OutputStreamType {
   }
 
   mutating func write(string: String) {
-    // FIXME: buffering?
-    // It is important that we use stdio routines in order to correctly
-    // interoperate with stdio buffering.
+    if string.isEmpty { return }
+
+    if string._core.isASCII {
+      defer { _fixLifetime(string) }
+
+      let result = _swift_stdlib_fwrite_stdout(
+        UnsafePointer(string._core.startASCII), string._core.count, 1)
+      if result != 1 {
+        fatalError("fwrite() returned \(result), expected 1")
```

Comment:
on Tue Mar 01 2016, Dmitri Gribenko <notifications-AT-github.com> wrote:

>> @@ -330,9 +330,19 @@ internal struct _Stdout : OutputStreamType {
>>    }
>>  
>>    mutating func write(string: String) {
>> -    // FIXME: buffering?
>
>> -    // It is important that we use stdio routines in order to correctly
>> -    // interoperate with stdio buffering.
>> +    if string.isEmpty { return }
>> +
>> +    if string._core.isASCII {
>> +      defer { _fixLifetime(string) }
>> +
>> +      let result = _swift_stdlib_fwrite_stdout(
>> +        UnsafePointer(string._core.startASCII), string._core.count, 1)
>> +      if result != 1 {
>> +        fatalError("fwrite() returned \(result), expected 1")
>
> I'd say that putchar version is buggy.  Failure to print is a data loss, and we don't tolerate those in Swift.

Hmm, I'm not entirely sure that print shouldn't be considered more
casually than that by default.

-- 
-Dave